src/main/scala/org/dberg/hubot/listeners/TestListener3.scala

```diff
@@ -0,0 +1,21 @@
+package org.dberg.hubot.listeners
+
+import org.dberg.hubot.Hubot
+import org.dberg.hubot.listeners.Listener.CallbackSuccess
+import org.dberg.hubot.models._
+
+class TestListener3(hubot: Hubot) extends Listener(hubot) {
+
+  val callback: Callback = {
```

Comment:
In that case, there's no reason to have `DirectMessage` and `MessageType.DirectMessage`. I would just mov the unapply to the DirectMessage and GroupMessage objects. And then maybe trim the `Message` suffix off in that case since it's unambiguous ("Message" is already in `MessageType` and `MessageType.GroupMessage` is a mouthful)?

```scala
  object MessageType {
   case object Direct extends MessageType {
     def unapply(message: Message) = message.messageType match {
       case DirectMessage => true
       case _ => false
     }
   }
   case object Group extends MessageType {
     def unapply(message: Message) = message.messageType match {
       case GroupMessage => true
       case _ => false
     }
   }
  }
....
// which allows the less verose
case MessageType.Group() ..... =>
...
```